### ISSUE TITLE
Format="0" when provisioning CalculatedField

### ISSUE THREAD

**Issue by esuhendra** [SubPointSolutions/spmeta2#969]
Hi,

Having some problem deploying Calculated field using SPMeta2 CSOM. I can set the OutputType to DateTime but for some reason when I interrogate the Schema XML of the Calculated field, it's showing Format="0". I'm pretty sure that the CalculatedFieldDefinition already setting the DateFormat to be DateOnly, but for some reason it's not setting the Format attribute. 

Expected Field Schema XML:
`<Field ID="{c1fdb24d-0e4a-494e-ba53-642b5e9e64a2}" StaticName="Sortable" DisplayName="Sortable" Title="Sortable" Name="Sortable" Type="Calculated" Group="Custom Columns" CanToggleHidden="TRUE" Hidden="FALSE" Indexed="FALSE" LCID="1033" **Format="DateOnly"** ResultType="Text" ...
`

Actual Field Schema XML:
`<Field ID="{c1fdb24d-0e4a-494e-ba53-642b5e9e64a2}" StaticName="Sortable" DisplayName="Sortable" Title="Sortable" Name="Sortable" Type="Calculated" Group="Custom Columns" CanToggleHidden="TRUE" Hidden="FALSE" Indexed="FALSE" LCID="1033" **Format="0"** ResultType="Text" ...
`
Not sure if it's only happening to me?
I'm using SPMeta2 v1.2.80 CSOM

**Comment by SubPointSupport** [comment]
It's a bit messy but SharePoint API exposed the following props for calculated field:
* DateFormat prop -> Format xml attribute
* OutputType prop -> ResultType xml attribute

Next, from the code inspection we see the following behavior:
1) new field creation - both DateFormat/OutputType  are used to form field XML
2) field update - only OutputType prop is used to update existing field

We'll have a close look on why DateFormat isn't used while updating the field. It might be lack of CSOM api, it might be only available in the recent CUs or O365 api or, surely, might be our bug in SPMeta2.

Moving forward, would you pls share (1) your field definition and (2) the following trace:
```cs
var info = SPMeta2Diagnostic.GetDiagnosticInfo();
```

Once done, we'll work it out for you.

**Comment by esuhendra** [comment]
Sure.

Here's my field definition:
```
var defn = new CalculatedFieldDefinition()
{
Id = new Guid("{c1fdb24d-0e4a-494e-ba53-642b5e9e64a2}"),
Title = "Sortable",
InternalName = "Sortable",
Group = "Custom Column",
FieldType = "Calculated",
Formula = "=[{0}]&amp;" : "&amp;[{1}]",
FieldReferences = new Collection<string>(fldRefs.ToList()),
OutputType = "Text",
DateFormat = BuiltInDateTimeFieldFormatType.DateOnly,
};
```

And the diagnostic info requested:
```
SPMeta2.Diagnostic.DiagnosticInfo SPMeta2FileVersion:[1.2.17057.1048] SPMeta2FileLocation:[E:\Development\Trunk\SharePointOnline\NET.SPProvisionTool\NET.SPProvisionForm\bin\Debug\SPMeta2.dll] CSOMFileVersion:[16.1.3912.1204] CSOMFileLocation:[E:\Development\Trunk\SharePointOnline\NET.SPProvisionTool\Bluebox.NET.SPProvisionForm\bin\Debug\Microsoft.SharePoint.Client.dll] Convert(p.IsSSOMDetected):[False]
```

I have also updated the SPMeta2 package to v1.2.110 and still the same issue.

**Comment by SubPointSupport** [comment]
Sweet. Let us work it out, will get back in a day or so.

**Comment by SubPointSupport** [comment]
Fixed, will push to NuGet as 1.2.120 soon.

**Comment by esuhendra** [comment]
Thanks guys!
